### ISSUE TITLE
a11y: emoji panel not useable with keyboard

### ISSUE THREAD

**Issue by pvagner** [deltachat/deltachat-desktop#5602]
- Operating System (Linux/Mac/Windows/iOS/Android): tested on linux but should be the same on other platforms
- Delta Chat Version: 2.22
- Expected behavior: I'd like to be able to use arrow keys to select individual emojis when emoji panel is visible after pressing emoji button while composing message
- Actual behavior: On linux with electron flavor and orca as a screen reader all the visible emojis are spoken when the grid with emojis receives the focus and I am even unsure if pressing arrow keys in deed does something usefull. With tauri flavor it's not accessible at all.
- Steps to reproduce the problem:
 * Open a chat and use tab key to navigate until the Emoji button receives the focus and activate it
 * Search entry receives the focus. Make sure the screen reader is switched into focus mode and Press the tab key to move to the grid with the emojis
 * Try to select one by using all the arrow keys.

**Comment by WofWca** [comment]
> Press the tab key to move to the grid with the emojis

The idea is to use arrow keys while the search input is focused. This way it should work, and does announce the selected one, at least with NVDA.

**Comment by pvagner** [comment]
Hello,

Oh in this case this turned out to be my problem not discovering how this feature is supposed to work. So now I see this is technically accessible but a future improvement might be trying to make it more discoverable. Again this is my fault not looking into it through DOM explorer.

The first thing I don't know is a feature or an issue is that pressing the tab key while the emoji search field is focused moves focus to the parent element of the emojis. This currently looks missleading to me.

To help discovering I should use arrow keys to navigate while the emoji search entry has focus it might be a good idea to add aria-autocomplete="list" in to that input field.
Here is the MDM documentation on that attribute: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-autocomplete

Now looking at this using the dev tools it looks as the reusable component and I am not sure we have this under our control.
The markup of the emoji search field looks like this:
```
<input type="search" placeholder="Search" autocomplete="off">
```
My interpretation is that autocomplete="off" is telling the renderer engine (electron in this case) not to provide assistance while filling this input entry. And aria-autocomplete might still be usefull.
https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/autocomplete

Oh with tauri on linux using arrow keys to navigate is not working for me with this input field at all. But I am seeing other weird things with webkit-gtk on linux lately including unexpected focus handling, different role mappings, not handling multi line content, so I am no longer seeing running tauri edition on linux viable alternative to electron.

**Comment by nicodh** [comment]
As I understand this it basically works but there is place for improvements. As the emoji picker is a third party library we have not much possibilities here to change markup.

**Comment by pvagner** [comment]
Thanks @nicodh for checking this.

Now the interaction works like this:

* Press the Smileys button
* Search entry receives the focus
* Searching and using arrow keys to navigate is working fine that's the way on how to use it
* Pressing the tab key moves focus to the grid which is not actionable by using the keyboard and is somewhat missleading. If this part is controlled by the library deltachat is using as a dependency perhaps I have to report to the library maintainers. I'll look into it if I can find it used elsewhere and identify the same thing.

Huge thanks once again.

**Comment by nicodh** [comment]
> If this part is controlled by the library deltachat is using as a dependency perhaps I have to report to the library maintainers. I'll look into it if I can find it used elsewhere and identify the same thing.

The library we used was not maintained since years https://github.com/missive/emoji-mart so we just switched to a fork that had at least some updates. But I didn't found a fork that was maintained by a real community - only single maintainer forks. We don't want to become a maintainer for the library ourselves since we already have enough stuff to maintain.

If you find a fork that has better accessibility support we are interested tohear about it.

**Comment by nicodh** [comment]
btw @pvagner I tried to contact you via deltacht - can you contact me please on this temporary profile? https://i.delta.chat/#8BEAEAA4D0F9DF5C1BDB21A338179E5C1AC15D7C&v=3&i=tV5jz7-8z4Yj8seBDlfkyWS6&s=rXXt1fsQSv5M0cSbMbSDNqsP&a=dpk6wdwmw%40nine.testrun.org&n=nicodh
